draft-ietf-httpbis-early-hints.md

```diff
@@ -90,21 +90,22 @@ The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "S
 The 103 (Early Hints) informational status code indicates to the client that the server is likely to
 send a final response with the header fields included in the informational response.
 
-A server MUST NOT include Content-Length, Transfer-Encoding, or any hop-by-hop header fields
-([RFC7230], Section 6.1) in a 103 (Early Hints) response.
-
 A client can speculatively evaluate the header fields included in a 103 (Early Hints) response while
 waiting for the final response. For example, a client might recognize a Link header field value
 containing the relation type "preload" and start fetching the target resource.
-
 However, these header fields only provide hints to the client; they do not replace the header
-fields on the final response. Aside from performance optimizations, such evaluation of the 103
+fields on the final response.
+
+Aside from performance optimizations, such evaluation of the 103
 (Early Hints) response's header fields MUST NOT affect how the final response is processed. A
 client MUST NOT interpret the 103 (Early Hints) response header fields as if they applied to
 the informational response itself (e.g., as metadata about the 103 (Early Hints) response).
 
-An intermediary MAY drop the informational response. It MAY send HTTP/2 ([RFC7540]) server pushes
-using the information found in the 103 (Early Hints) response.
+A server MUST NOT omit sending the header fields of a final response even if they have been sent as
+part of a 103 (Early Hints) response.
```

Comment:
What's the intent here -- to require headers in the 103 to be repeated in the final response? Editorial issue aside, that seems like it might always work; e.g., when the 103 headers are speculative.